test/test_calendar.py

```diff
@@ -150,3 +154,42 @@ def test_parse_date_offset_no_offset_magnitude():
 
     expected_error = "No numerical value given for offset: YS"
     assert str(exc_info.value) == expected_error
+
+
+@pytest.mark.parametrize(
+        "start_date, end_date, caltype_int, expected",
+        [
+            (
+                datetime.datetime(year=4, month=1, day=1),
+                datetime.datetime(year=5, month=1, day=1),
+                GREGORIAN,
+                366 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=4, month=1, day=1),
+                datetime.datetime(year=5, month=1, day=1),
+                NOLEAP,
+                365 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=300, month=1, day=1),
+                datetime.datetime(year=301, month=1, day=1),
+                GREGORIAN,
+                365 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=400, month=1, day=1),
+                datetime.datetime(year=401, month=1, day=1),
+                GREGORIAN,
+                366 * SEC_PER_DAY
+            ),
+            (
+                datetime.datetime(year=1, month=1, day=1),
+                datetime.datetime(year=1001, month=1, day=1),
+                GREGORIAN,
+                (1000 * 365 + 242) * SEC_PER_DAY
+            ),
+        ]
+)
```

Comment:
Great to add tests, thanks. Just a note that all of the tests use `month=1` and `day=1`, so probably worth changing a couple to something else.

Reply:
Good idea. I've added in tests using different months and days, and also added a test running between year 1 and 9999, which are the minimum and maximum years for `datetime.date`s.

Reply:
Thanks - new test looks good :)